specreduce/core.py

```diff
@@ -2,3 +2,246 @@
+# import scipy.signal
+from scipy.interpolate import UnivariateSpline
+# from scipy.interpolate import BSpline, SmoothBivariateSpline
+
+# from specutils import SpectralRegion
+
+__all__ = ["genfluxcal", "applyfluxcal"]
+
+
+def _mag2flux(wave, mag, zeropoint=48.60):
+    """
+    An internal method to convert quantities in magnitudes to flux units.
+    Important for handling standard stars as well as other data files ported
+    from IRAF, such as the data files in the specreduce/datasets/onedstds
+    directory. Basically converts to "PHOTFLAM" units. Assumes that
+    the common flux zero point as used in IRAF.
+
+    Parameters
+    ----------
+    wave : a 1D numpy array
+        The wavelength of the data points.
+
+    mag : a 1D numpy array
+        The magnitudes of the data points.
+
+    zeropoint : float, optional
+        Default is set to 48.60. Conversion factor for converting mag into flux.
+
+    Returns
+    -------
+    Corresponding quantity in flux units.
+    """
+
+    c = 2.99792458e18  # speed of light, in Angstrom/second
+    flux = 10.0 ** ((mag + zeropoint) / (-2.5))
+    return flux * (c / wave ** 2.0)
+
+
+def genfluxcal(obj_wave, obj_flux, std, mode="spline", polyorder=9, show=False):
+    """
+    Generate the sensitivity function (or sensfunc) needed for the flux calibration of
+    a 1D spectrum.
+
+    Parameters
+    ----------
+    obj_wave : a 1D numpy array
+        The 1D wavelength array of the spectrum to be flux calibrated.
+
+    obj_flux : a 1D numpy array
+        The 1D flux array of the spectrum to be flux calibrated.
+
+    std : str
+        Name of the standard star file in the specreduce/datasets/onedstds
+        directory to be used for the flux calibration. The user must provide the
+        subdirectory and file name. For example:
+
+        >>> genfluxcal(obj_wave, obj_flux, std='/ctiocal/eg21', mode='poly', polyorder=6)  \
+        # doctest: +SKIP
+
+        If no std is supplied, will raise a ValueError.
+
+    mode : str, optional
+        Default to 'spline'. Can also be set to either 'linear' or 'poly'.
+
+    polyorder: float, optional
+        Default to '6'. Is defined as the order of the polynomial for the 'poly' fitting.
+        Only needed if the 'poly' mode has been chosen.
+
+    show: bool, optional
+        Default to 'False'. If set to 'True', shows a plot of the sensfunc determined.
+
+    Returns
+    -------
+    sensfunc : a 1D numpy array
+        The sensfunc derived with the standard star for the flux calibration.
+
+    """
+
+    if std is None:
+        msg1 = "Error: No 'onedstd' standard star file has been specified." \
+               "Cannot proceed further."
+        raise ValueError(msg1)
+
+    std = std.lower()
+    std_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           "datasets", "onedstds")
+
+    if os.path.isfile(os.path.join(std_dir, std)):
+        std_wave, std_mag, std_wth = np.genfromtxt(
```

Comment:
I've been switching to using astropy tables for reading these onedstds in, hoping that's more stable and useful?

Reply:
See comment above, but could these also be read in with the specutils functionality?

Reply:
Let me try this out and compare with other methods...

Reply:
Yeah, could compare different approaches say with using `astropy.table` or even some similar `specutils` and refactor this code snippet and see which one is more suitable for the context.